include/ArcadeMachine.h

```diff
@@ -20,333 +20,467 @@
+        {
+            write_line("Exit button clicked");
+            exit_program();
+        }
+    }
+
+    /**
+        Draws the Main Menu
+    */
+    void draw_main_menu() 
+    {
+        // Get mouse position
+        this->_mouse = mouse_position();
+        this->_grid.DrawGrid();
+        // Draw cursor
+        draw_sprite(this->_selector_main_menu.get_cursor());
+        // Get button postions
+        cell play = this->_grid.GetCell(2, 10);
+        cell options = this->_grid.GetCell(3, 10);
+        cell exit = this->_grid.GetCell(4, 10);
+        // Arcade Machine title
+        draw_text("Arcade",  COLOR_BLACK, "font_title", 100, 1180, 100);
+        draw_text("Machine", COLOR_BLACK, "font_title", 100, 1150, 200);
+        // Draw text on buttons
+        draw_text("play!", COLOR_BLACK, "font_btn", 70, play.button->x() + (play.button->centre_x()/2) + 5, play.button->y() + 5);
+        draw_text("options", COLOR_BLACK, "font_btn", 70, options.button->x() + (options.button->centre_x()/2) - 20, options.button->y() + 5);
+        draw_text("exit", COLOR_BLACK, "font_btn", 70, exit.button->x() + (exit.button->centre_x()/2) + 20, exit.button->y() + 5);
+
+        // Check input in selector class.
+        this->menu_button_node = this->_selector_main_menu.check_key_input(this->menu_button_node);
+
+        // Check input in selector class.
+        this->_action = this->_selector_main_menu.check_for_selection(this->menu_button_node);
+    }
+
+    /**
+        Prepares the Main Menu
+    */
+    void prepare_main_menu()
+    {
+        // Get the data from the config files.
+        this->_configs = this->_helper.config_data_list();
+        // Initialise grid 
+        Grid grid(ROWS, COLS);
+        this->_grid = grid;
+        // Create menu buttons
+        Button *play = new MenuButton(Button::PLAY, 1.5);
+        Button *opts = new MenuButton(Button::OPTS, 1.5);
+        Button *exit = new MenuButton(Button::EXIT, 1.5);
+
+        // Add menu buttons to local vector
+        this->_menu_btns.push_back(play);
+        this->_menu_btns.push_back(opts);
+        this->_menu_btns.push_back(exit);
+
+        // Fetch menu background
+        bitmap thoth = bitmap_named("thoth");
+        // Update grid cells with assets
+        this->_grid.SetBackground(thoth);
+
+        // Make new nodes with buttons.
+        this->menu_button_node = new ButtonNode(play);
+        this->menu_button_node->addAfter(new ButtonNode(opts));
+        this->menu_button_node->addBefore(new ButtonNode(exit));
+
+        // Update grid with nodes.
+        this->_grid.UpdateCell(menu_button_node->button, 2, 10);
+        this->_grid.UpdateCell(menu_button_node->getPrev()->button, 4, 10);
+        this->_grid.UpdateCell(menu_button_node->getNext()->button, 3, 10);
+
+        // Play main menu music
+        if (this->_play_music) play_music("music_mainmenu");
+    }
+
+    /**
+        DEPRACATED
```

Comment:
Not sure why you would bother leaving this in if you've updated/refactored it? 1/2